hedera-mirror-monitor/src/main/java/com/hedera/mirror/monitor/publish/TransactionPublisher.java

```diff
@@ -137,19 +153,26 @@ public void close() {
     }
 
     private Flux<Client> getClients() {
+        Map<String, AccountId> validNodes = getValidNodes();
+        log.info("Creating {} connections to {} nodes", publishProperties.getClients(), validNodes.size());
+
+        return Flux.range(0, publishProperties.getClients())
+                .flatMap(i -> Flux.defer(() -> Mono.just(toClient(validNodes))));
+    }
+
+    private Map<String, AccountId> getValidNodes() {
+        log.info("Validating nodes");
         List<NodeProperties> validNodes = validateNodes();
 
         if (validNodes.isEmpty()) {
             throw new IllegalArgumentException("No valid nodes found");
         }
 
+        nodeAccountIds.clear();
         validNodes.forEach(n -> nodeAccountIds.add(AccountId.fromString(n.getAccountId())));
```

Comment:
I don't like having a period of time with zero nodeAccountIds. Let's change it to use `AtomicReference` so that doesn't occur. We can also get away with just using a regular `ArrayList` that way:

```suggestion
        List<AccountId> newNodeAccountIds = new ArrayList<>();
        validNodes.forEach(n -> newNodeAccountIds.add(AccountId.fromString(n.getAccountId())));
        nodeAccountIds.set(newNodeAccountIds);
```

Reply:
Good idea, implemented this.